packages/daemon/src/pet-sitter.js

```diff
@@ -47,16 +47,35 @@ export const makePetSitter = (petStore, specialNames) => {
   const list = () =>
     harden([...Object.keys(specialNames).sort(), ...petStore.list()]);
 
-  /** @type {import('./types.js').PetStore['follow']} */
-  const follow = async function* currentAndSubsequentNames() {
+  /** @type {import('./types.js').PetStore['followNameChanges']} */
+  const followNameChanges = async function* currentAndSubsequentNames() {
     for (const name of Object.keys(specialNames).sort()) {
       const idRecord = idRecordForName(name);
       yield /** @type {{ add: string, value: import('./types.js').IdRecord }} */ ({
         add: name,
         value: idRecord,
       });
     }
-    yield* petStore.follow();
+    yield* petStore.followNameChanges();
+  };
+
+  /** @type {import('./types.js').PetStore['followIdNameChanges']} */
+  const followIdNameChanges = async function* currentAndSubsequentIds(id) {
+    const subscription = petStore.followIdNameChanges(id);
+
+    const [idSpecialName] = Object.entries(specialNames)
+      .filter(([_, specialId]) => specialId === id)
+      .map(([specialName, _]) => specialName);
```

Comment:
We don’t currently assign multiple special names to an identifier, but I think we can eliminate the `if` block below if we assume there can be zero or more entries in this array.